### DISCUSSION TITLE
How to revalidate server side component in next js 13

### DISCUSSION THREAD

**Discussion by LouisXDD** [vercel/next.js#54479] [App Router]
I am going develop a ecommerce. And I created a search page and product detail page.
But hen I go to the product detail page once time, there are never refresh again, however I change the product info from db, there are nothing chagne.

**Comment by hyjoong** [comment]


**Reply by milovangudelj** [reply]
This might not always be the best option. If the data doesn't really change that often on-demand cache revalidation might be a more appropriate approach.

https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#on-demand-revalidation

**Reply by burhansancakli** [reply]
99% of users don't care about these kind of small optimizations, and those who care already know about it as they actually read the documentations.

**Reply by milovangudelj** [reply]
You would be surprised knowing how many people complain about this and don't read the docs about it. It's a lot.

Also, your comment doesn't add anything valuable to the discussion. 😅

**Comment by burhansancakli** [comment]
Nextjs tries to cache everything it can by default. You can completely disable this for specific components and routes.

If you're on a component, you can change your fetch function and add "no-cache" like this:
`fetch('https://...', { cache: 'no-store' })`

If you're on the API Route, you can export a const like this:
`export const dynamic = 'force-dynamic'`

**Comment by AnastasiiaRoch** [comment]
Try the on-demand revalidation using the [revalidateTag](https://nextjs.org/docs/app/api-reference/functions/revalidateTag) method. [Here](https://medium.com/stackademic/how-to-revalidate-server-components-in-next-js-13th-a4ee49cd2c8a) is an example
